docs/data/data-grid/list-view/ListViewAdvanced.tsx

```diff
@@ -0,0 +1,362 @@
+        type: fileType as FileType,
+        size: file.size,
+        createdBy: 'Kenan Yusuf',
+        createdAt,
+        updatedAt: createdAt,
+        state: 'pending',
+      };
+
+      event.target.value = '';
+
+      // Add temporary row
+      setLoading(true);
+      setRows((prevRows) => [...prevRows, row]);
+
+      // Simulate server response time
+      setTimeout(() => {
+        const uploadedRow: RowModel = { ...row, state: 'uploaded' };
+        setRows((prevRows) =>
+          prevRows.map((r) => (r.id === row.id ? uploadedRow : r)),
+        );
+        setDetailsState({ open: true, params: { row } });
+        setLoading(false);
+      }, 2000);
+    },
+    [],
+  );
+
+  const columns: GridColDef[] = React.useMemo(
+    () => [
+      {
+        field: 'name',
+        headerName: 'Name',
+        width: 350,
+        editable: true,
+        hideable: false,
+        renderCell: (params) => {
+          return (
+            <Stack
+              direction="row"
+              gap={1.5}
+              alignItems="center"
+              sx={{ height: '100%' }}
+            >
+              <FileIcon type={params.row.type} />
+              {params.value}
+            </Stack>
+          );
+        },
+      },
+      {
+        field: 'createdBy',
+        headerName: 'Owner',
+        width: 200,
+        renderCell: (params) => {
+          const avatarProps = stringAvatar(params.value);
+          return (
+            <Stack direction="row" gap={1.5} alignItems="center">
+              <Avatar
+                {...avatarProps}
+                sx={{ width: 24, height: 24, fontSize: 12, ...avatarProps.sx }}
+              />
+              {params.value}
+            </Stack>
+          );
+        },
+      },
+      {
+        field: 'createdAt',
+        headerName: 'Added',
+        type: 'date',
+        width: 200,
+        valueFormatter: formatDate,
+      },
+      {
+        field: 'updatedAt',
+        headerName: 'Modified',
+        type: 'date',
+        width: 200,
+        valueFormatter: formatDate,
+      },
+      {
+        field: 'type',
+        headerName: 'Type',
+        width: 150,
+      },
+      {
+        field: 'size',
+        headerName: 'Size',
+        width: 120,
+        valueFormatter: formatSize,
+      },
+      {
+        type: 'actions',
+        field: 'actions',
+        resizable: false,
+        width: 50,
+        getActions: (params) =>
+          isListView
+            ? [
+                <ActionDrawer
+                  params={params}
+                  container={container}
+                  onSaveRename={(value) => handleUpdate(params.id, 'name', value)}
+                  onDelete={() => handleDelete([params.id])}
+                  onOpen={() => {
+                    setDetailsState({ open: true, params });
+                  }}
+                />,
+              ]
+            : [
+                <GridActionsCellItem
+                  label="Open"
```

Comment:
Nitpick: Since the click is supposed to show the information about file not actually Open it.
```suggestion
                  label="Preview"
```

Reply:
Good shout, updated here https://github.com/mui/mui-x/pull/14393/commits/8933fd4c4f110c09f90fe70b2f6d5a0396be9030